### PR TITLE
BZ #1180158: Ceilometer workload partitioning with tooz & redis

### PR DESCRIPTION
**Pull request by eglynn**
https://bugzilla.redhat.com/show_bug.cgi?id=1180158

The quickstack::pacemaker:ceilometer puppet class now allows
redis to be specified as the backend for tooz, to be used for
workload partitioning in the ceilometer central agent and
alarm evaluator.

We do not create a pacemaker resource for redis, as this will
instead be self-monitored via the redis-sentinel service (via
a subsequent patch, once the support for sentinel in tooz is
packaged python-tooz).

### CODE REVIEW

**Comment by eglynn** [comment]
Capturing some relevant discussion from IRC ...

```
 <jayg> eglynn-office: before I forget, since convo is moved to your new PR, I like your thought of the vip +haproxy on the sentinels, that may solve the part I was concerned about
 <eglynn-office> jayg: thanks ... a potentially even simpler approach is being proposed to tooz by cdent here https://review.openstack.org/146463
 <eglynn-office> jayg: ... i.e. allowing *all* of the sentinels to be identified in the backend URL
 <jayg> eglynn-office: sure, that works too, more like we do with mongo already
 <eglynn-office> jayg: coolness
```

File: puppet/modules/quickstack/manifests/pacemaker/ceilometer.pp [hunk 1]
```diff
@@ -1,8 +1,12 @@
 class quickstack::pacemaker::ceilometer (
   $ceilometer_metering_secret,
-  $memcached_port          = '11211',
-  $db_port                 = '27017',
-  $verbose                 = 'false',
+  $memcached_port                = '11211',
+  $db_port                       = '27017',
+  $verbose                       = 'false',
+  $coordination_backend          = 'redis',
```
Comment (jguiditta) [inline puppet/modules/quickstack/manifests/pacemaker/ceilometer.pp]: Super minor nit - could you put these new params in alphabetical order? I think it is easier to find things that way

**Comment by jguiditta** [comment]
Minor nit aside, is this ready for me to test against release repos?  If so, and we have all needed acks, we can try to include it tomorrow

**Comment by eglynn** [comment]
@jguiditta: to be fully functional, the [second commit](https://github.com/eglynn/astapor/commit/195822ac) of this pull request requires new builds of openstack-puppet-modules and python-tooz. After discussions around the o-p-m rebuild yesterday, it was decided by eng mgmt not to incur the risk of rebuilding o-p-m prior to the OSP6 release candidate.

The [first commit](https://github.com/eglynn/astapor/commit/f77577ea) on the other hand, should work against the release repos, as it does not install sentinel (the aspect for which the new build of o-p-m would be required).

From our discussion on the [original pull-request](https://github.com/redhat-openstack/astapor/issues/447), I suspect that you may be leery of only including the basic redis support without sentinel in OSP6. However if that kind of incremental approach is indeed a runner, I could recreate the pull request with only the initial commit.

**Comment by jguiditta** [comment]
Closing this, as the functionality is included in #506

**Pull request closed without merging**